sklearn/decomposition/_dict_learning.py

```diff
@@ -1630,6 +1631,33 @@ class DictionaryLearning(_BaseSparseCoding, BaseEstimator):
     0.07...
     """
 
+    _parameter_constraints = {
+        "n_components": [Interval(Integral, 0, None, closed="neither"), None],
+        "alpha": [Interval(Real, 0, 1, closed="left")],
+        "max_iter": [Interval(Integral, 1, None, closed="neither")],
+        "tol": [Interval(Real, 0, None, closed="left")],
+        "fit_algorithm": [StrOptions(
+            {"lars", "cd"})
+        ],
+        "transform_algorithm": [
+            StrOptions({"lasso_lars", "lasso_cd", "lars", "omp", "threshold"})
+        ],
+        "transform_n_nonzero_coefs": [
+            Interval(Integral, 0, None, closed="neither"),
+            None
+        ],
+        "transform_alpha": [Interval(Real, 0, 1, closed="left"), None],
+        "n_jobs": [Interval(Integral, -1, None, closed="neither"), None],
+        "code_init": ["array-like", None],
+        "dict_init": ["array-like", None],
```

Comment:
Could you also update the documentation of these two parameters to reflect that they may be `array-like` and not strictly `np.ndarray`?

Reply:
I think a ndarray is better for now because we don't validate these so passing like a list would break in a non-informative way at some point.